packages/opentelemetry-instrumentation-grpc/src/grpc-js/clientUtils.ts

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GrpcJsInstrumentation } from './';
+import type { GrpcClientFunc, SendUnaryDataCallback } from './types';
+import {
+  SpanKind,
+  Span,
+  SpanStatusCode,
+  SpanStatus,
+  propagation,
+  context,
+} from '@opentelemetry/api';
+import { RpcAttribute } from '@opentelemetry/semantic-conventions';
+import type * as grpcJs from '@grpc/grpc-js';
+import {
+  _grpcStatusCodeToSpanStatus,
+  _grpcStatusCodeToOpenTelemetryStatusCode,
+  _methodIsIgnored,
+} from '../utils';
+import { CALL_SPAN_ENDED } from './serverUtils';
+import { EventEmitter } from 'events';
+
+/**
+ * Parse a package method list and return a list of methods to patch
+ * with both possible casings e.g. "TestMethod" & "testMethod"
+ */
+export function getMethodsToWrap(
+  this: GrpcJsInstrumentation,
+  client: typeof grpcJs.Client,
+  methods: { [key: string]: { originalName?: string } }
+): string[] {
+  const methodList: string[] = [];
+
+  // For a method defined in .proto as "UnaryMethod"
+  Object.entries(methods).forEach(([name, { originalName }]) => {
```

Comment:
FYI: While I don't believe that this is an issue for this package, Object.entries() is not supported in any IE versions including IE 11.

Reply:
No its not possible to this into the browser since we need to hook into the `require` system for those instrumentations. Also i don't think that you can use `grpc-js` as-is inside browser ?

Reply:
this is node package only